app/vlinsert/datadog/datadog.go

```diff
@@ -61,15 +61,14 @@ func datadogLogsIngestion(w http.ResponseWriter, r *http.Request) bool {
 		ts = startTime.UnixNano()
 	}
 
-	if r.Header.Get("Content-Encoding") == "gzip" {
-		zr, err := common.GetGzipReader(reader)
-		if err != nil {
-			httpserver.Errorf(w, r, "cannot read gzipped logs request: %s", err)
-			return true
-		}
-		defer common.PutGzipReader(zr)
-		reader = zr
+	encoding := r.Header.Get("Content-Encoding")
+	zr, err := common.GetUncompressedReader(reader, encoding)
```

Comment:
it's better for simplicity to rewrite this line to:
`reader, err := common.GetUncompressedReader(r.Body, encoding)`